src/server/engine_shard_set.cc

```diff
@@ -167,27 +167,23 @@ class RoundRobinSharder {
 
 bool HasContendedLocks(unsigned shard_id, Transaction* trx, const DbTable* table) {
   bool has_contended_locks = false;
+  auto record = [table, &has_contended_locks](string_view key) {
```

Comment:
i personally prefer to reduce captures to minimum. here I think it make sense to return the bool result instead of setting `has_contended_locks`. also it will allow you to break from the loops